gen/generate_test.go

```diff
@@ -395,3 +422,58 @@ func TestThriftPackageImporter(t *testing.T) {
 		}
 	}
 }
+
+type serviceRequestMatcher struct {
+	req *api.GenerateServiceRequest
+}
+
+var _ gomock.Matcher = serviceRequestMatcher{}
+
+func newServiceRequestMatcher(req *api.GenerateServiceRequest) gomock.Matcher {
+	return &serviceRequestMatcher{
+		req: req,
+	}
+}
+
+func (s serviceRequestMatcher) Matches(x interface{}) bool {
+	req, ok := x.(*api.GenerateServiceRequest)
+	if !ok {
+		return false
+	}
+
+	var intsA []int
+	for _, mID := range s.req.RootModules {
+		intsA = append(intsA, int(mID))
+	}
+
+	var intsB []int
```

Comment:
if we're not matching other fields of GenerateServivceRequest, perhaps this should be `RootModulesMatcher`?

In fact, it can probably be `type rootModulesMatcher []api.ModuleID`.